composer/algorithms/selective_backprop/selective_backprop.py

```diff
@@ -213,15 +213,23 @@ def match(self, event: Event, state: State) -> bool:
         )
         return is_chosen
 
-    def apply(self, event: Event, state: State, logger: Optional[Logger] = None) -> None:
+     def apply(self, event: Event, state: State, logger: Optional[Logger] = None, input_pos: Union[int, str, list, None] = None, target_pos: Union[int, str, list, None] = None) -> None:
```

Comment:
Suggest moving these to the algorithm's `__init__` signature, and then accessing here via attributes. Can also reduce the verbosity by defining a type alias.

```
def __init__(..., input_pos: Position = None, target_pos: Position = None):
    self.input_pos = input_pos
    self.target_pos = target_pos
```
cc: @eracah as FYI as well.

Reply:
Actually, this change seems out of scope with the PR's intent to simply change the defaults? Consider staging this into two PRs, one to simply change the defaults, and another to generalize to multiple batch types.

Reply:
yeah I think we should avoid adding code that tries to handle different types of batches on a per-algorithm basis until we resolve [this issue](https://mosaicml.atlassian.net/browse/CO-2).

Reply:
we will add some helper getters and setters and custom types that will make the specifying of which portion of the batch to grab easier for algorithm-writers.